packages/core/src/queries/log.ts

```diff
@@ -1,42 +1,72 @@
-import type { HookExecutionStats, Log } from '@logto/schemas';
-import { token, Logs } from '@logto/schemas';
+import {
+  token,
+  type hook,
+  Logs,
+  type HookExecutionStats,
+  type Log,
+  type interaction,
+} from '@logto/schemas';
 import { conditionalSql, convertToIdentifiers } from '@logto/shared';
+import { conditionalArray } from '@silverhand/essentials';
 import { subDays } from 'date-fns';
-import type { CommonQueryMethods } from 'slonik';
 import { sql } from 'slonik';
+import type { CommonQueryMethods } from 'slonik';
 
 import { buildFindEntityByIdWithPool } from '#src/database/find-entity-by-id.js';
 import { buildInsertIntoWithPool } from '#src/database/insert-into.js';
 
 const { table, fields } = convertToIdentifiers(Logs);
 
+type AllowedKeyPrefix = hook.Type | token.Type | interaction.Prefix;
+
 type LogCondition = {
   logKey?: string;
   applicationId?: string;
   userId?: string;
   hookId?: string;
   startTimeExclusive?: number;
+  includeKeyPrefix?: AllowedKeyPrefix[];
+  excludeKeyPrefix?: AllowedKeyPrefix[];
 };
 
 const buildLogConditionSql = (logCondition: LogCondition) =>
-  conditionalSql(logCondition, ({ logKey, applicationId, userId, hookId, startTimeExclusive }) => {
-    const subConditions = [
-      conditionalSql(logKey, (logKey) => sql`${fields.key}=${logKey}`),
-      conditionalSql(userId, (userId) => sql`${fields.payload}->>'userId'=${userId}`),
-      conditionalSql(
-        applicationId,
-        (applicationId) => sql`${fields.payload}->>'applicationId'=${applicationId}`
-      ),
-      conditionalSql(hookId, (hookId) => sql`${fields.payload}->>'hookId'=${hookId}`),
-      conditionalSql(
-        startTimeExclusive,
-        (startTimeExclusive) =>
-          sql`${fields.createdAt} > to_timestamp(${startTimeExclusive}::double precision / 1000)`
-      ),
-    ].filter(({ sql }) => sql);
+  conditionalSql(
+    logCondition,
+    ({
+      logKey,
+      applicationId,
+      userId,
+      hookId,
```

Comment:
nit: feels like could merge these three as a single `payload` parameter. So we could use conditionalArray on those Object.entries to build the SQL statement.